lib/screens/movie_list_home.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:flutter/material.dart';
+import '/movie_list_widgets/movie_list.dart';
+import '/repositories/movie_local_repository.dart';
+import '/repositories_i/repository_i.dart';
+
+import '/data_classes/movie.dart';
+
+class MovieListHome extends StatefulWidget {
+  const MovieListHome({super.key});
+
+  static const String routeName = "/";
+
+  @override
+  State<StatefulWidget> createState() => _MovieListHomeState();
+}
+
+class _MovieListHomeState extends State<MovieListHome> {
+  final IRepository movieRepository = MovieLocalRepository();
+  late Future<List<Movie>> movies;
+
+  @override
+  initState() {
+    super.initState();
+    movies = movieRepository.loadData() as Future<List<Movie>>;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: FutureBuilder<List<Movie>>(
+        future: movies,
+        builder: (
+          BuildContext context,
+          AsyncSnapshot<List<Movie>> snapshot,
+        ) {
+          if (snapshot.hasError) {
+            return Center(
+              child: Text(
+                "${snapshot.error}",
+              ),
+            );
+          } else if (snapshot.hasData) {
+            List<Movie> moviesList = snapshot.data!;
+            return MovieList(
+              moviesList: moviesList,
+            );
```

Comment:
You can use direclty snapshot.data! as a parameter of the MovieList

Reply:
fixed